src/observables/ForkJoinObservable.ts

```diff
@@ -3,41 +3,46 @@ import {Subscriber} from '../Subscriber';
 import {PromiseObservable} from './PromiseObservable';
 import {EmptyObservable} from './EmptyObservable';
 import {isPromise} from '../util/isPromise';
+import {isArray} from '../util/isArray';
 
 export class ForkJoinObservable<T> extends Observable<T> {
-  constructor(private sources: Array<Observable<any> |
-                                    Promise<any> |
-                                    ((...values: Array<any>) => any)>) {
-     super();
+  constructor(private sources: Array<Observable<any> | Promise<any>>,
+              private resultSelector?: (...values: Array<any>) => any) {
+    super();
   }
 
   static create(...sources: Array<Observable<any> |
+                                  Array<Observable<any>> |
                                   Promise<any> |
                                   ((...values: Array<any>) => any)>)
-                                  : Observable<any> {
-    if (sources === null || sources.length === 0) {
+                                  : Observable<any>;
+  static create(sources: Array<Observable<any> | Promise<any>>,
+                selector?: (...values: Array<any>) => any)
+                : Observable<any>;
+  static create(...sources: Array<any>): Observable<any> {
+    if (sources === null || arguments.length === 0) {
       return new EmptyObservable();
     }
-    return new ForkJoinObservable(sources);
-  }
 
-  private getResultSelector(): (...values: Array<any>) => any {
-    const sources = this.sources;
+    let resultSelector: (...values: Array<any>) => any = null;
+    if (typeof sources[sources.length - 1] === 'function') {
+      resultSelector = sources.pop();
```

Comment:
Appreciate for suggestion of mutating sources directly @blesh , this seems way better than I did originally. I should've tried to look existing references before submitting PR.